test/fw/ptl/utils/pbs_snaputils.py

```diff
@@ -0,0 +1,2361 @@
+
+        if "qmgr" in cmd[0]:
+            # qmgr -c is being called
+            if self.du.is_localhost(host) is False:
+                # For remote hosts, wrap qmgr's command in quotes
+                cmd[2] = "\'" + cmd[2] + "\'"
+
+        with open(out_path, "w") as out_fd:
+            try:
+                cmd_stdout = out_fd
+                if grep_pbs is True:
+                    # Set stdout to a temp file
+                    _, fn = self.du.mkstemp()
+                    tempfd = open(fn, "w")
+                    cmd_stdout = tempfd
+                self.du.run_cmd(host, cmd=cmd, stdout=cmd_stdout)
+                if grep_pbs is True:
+                    # Now run grep on the temp file and store output
+                    cmd_path = self.du.which(exe="grep")
+                    cmd = [cmd_path, "[p]bs", fn]
+                    cmd_stdout = out_fd
+                    self.du.run_cmd(cmd=cmd, stdout=cmd_stdout)
+                    # Close and delete the temp file
+                    tempfd.close()
+                    os.remove(fn)
+            except Exception, e:
+                self.logger.info(str(e))
+
+            if self.anonymize:
+                self.__anonymize_file(out_path)
+
+        if self.create_tar is True:
+            self.__add_to_archive(out_path)
+
+    def __initialize_server(self, hostname=None, diag=None):
+        """
+        Initialize a server instance. This can be done in one of
+        three ways:
+
+          1. query a PBS server by name.
+          2. read files from a pbs_diag directory
+          3. query information directly from the PBS database
+
+        :param hostname: the host to query
+        :type hostname: string or None
+        :param diag: the directory in which a pbs_diag is captured
+        :type diag: string or None
+        :returns: an instance of the class Server
+        """
+        server = Server(hostname, diag=diag)
+        if diag:
+            server.pbs_conf["PBS_HOME"] = diag
+            diag_time = self.get_diag_time(diag)
+            server.ctime = diag_time
+
+        if "PBS_SERVER" not in server.pbs_conf:
+            server.pbs_conf["PBS_SERVER"] = socket.gethostname()
+
+        if self.server_host is None:
+            self.server_host = server.hostname
+
+        return server
+
+    def __convert_flag_to_numeric(self, flag):
+        """
+        Convert a resource's flag attribute to its numeric equivalent
+
+        :param flag: the resource flag to convert
+        :type flag: string
+        :returns: numeric value of the resource flag
+        """
+        resc_flag = READ_WRITE
+        flag_ir = 0
+        if "q" in flag:
+            resc_flag |= ATR_DFLAG_RASSN
+        elif "f" in flag:
+            resc_flag |= ATR_DFLAG_FNASSN
+        elif "n" in flag:
+            resc_flag |= ATR_DFLAG_ANASSN
+        elif "h" in flag:
+            resc_flag |= ATR_DFLAG_CVTSLT
+        elif "r" in flag:
+            if flag_ir == 0:
```

Comment:
This was a loop at some point, I've removed flag_ir and converted the elifs to ifs. Please let me know if it looks fine now.